tests/simple_regular_exchange.cpp

```diff
@@ -41,7 +41,13 @@ struct memory
     unsigned int m_size;
     std::unique_ptr<T[]> m_host_memory;
 #ifdef __CUDACC__
-    struct cuda_deleter { void operator()(T* ptr) const {cudaFree(ptr);} };
+    struct cuda_deleter
+    {
+        // no delete since this messes up the rma stuff
```

Comment:
Is this a problem with any platform or only cray? When is the deletion happening?

Reply:
I tested this again, and the problem occurs only if I run 2 tests in a row (i.e. allocation - deallocation - allocation) with xpmem and gpu enabled.
I believe it is a cray problem, but I can't be a 100% sure. Have you got any idea how to test this further?